src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+
+        } else {
+            retryCount++;
+            if (retryCount < 5) {
+                Thread.sleep(10000);
+                logger.warn("Unable to connect to Biomart. Trying again...");
+                return queryBiomart(queryString, retryCount);
+            }
+            throw new Exception("Unable to connect to Biomart with URL: " + queryString);
+        }
+    }
+
+    /**
+     * Function that takes species name attribute from config file (eg: Homo sapiens) and modifies it to
+     * match Biomart formatting (first letter from primary species name + secondary name, all lowercase -- eg: hsapiens).
+     * @return List<String> of species names in Biomart format (eg: hsapiens).
+     * @throws IOException
+     * @throws ParseException
+     */
+    private List<String> getBiomartSpeciesNames() throws IOException, ParseException {
+        // Read properties file.
+        Properties applicationProps = new Properties();
+        String propertiesLocation = System.getProperty("config.location");
+        try(FileInputStream fis = new FileInputStream(propertiesLocation))
+        {
+            applicationProps.load(fis);
+        }
+        // Get species JSON config file location, import as JSON object.
+        String pathToSpeciesConfig = applicationProps.getProperty("pathToSpeciesConfig");
+        JSONParser parser = new JSONParser();
+        Object obj = parser.parse(new FileReader(pathToSpeciesConfig));
+        JSONObject jsonObject = (JSONObject) obj;
+
+        // For each species in config file, retrieve name, modify it and then add to list.
+        List<String> biomartNames = new ArrayList<>();
+        for (Object speciesKey : jsonObject.keySet()) {
+            JSONObject speciesJson = (JSONObject) jsonObject.get(speciesKey);
+            JSONArray speciesNames = (JSONArray) speciesJson.get("name");
+            String speciesName = (String) speciesNames.get(0);
+            String biomartName = speciesName.substring(0,1).toLowerCase() + speciesName.split(" ")[1];
+            biomartNames.add(biomartName);
+        }
+        return biomartNames;
+    }
+
+    // Gets the filepath to biomart XML file, used to build biomart query.
+    private String getBiomartXMLPath() throws IOException {
+        // Read properties file.
+        Properties applicationProps = new Properties();
+        String propertiesLocation = System.getProperty("config.location");
+        try(FileInputStream fis = new FileInputStream(propertiesLocation))
+        {
+            applicationProps.load(fis);
+        }
+        // Get species JSON config file location, import as JSON object.
+        return applicationProps.getProperty("pathToBiomartXML");
+    }
+
+    // Modifies default secondary URL with species name and the query type (for example microarray probe type, or uniprotsptrembl).
+    private String getBiomartIdentifierQuery(String pathToXMLQuery, String biomartSpeciesName, String queryId) throws IOException {
```

Comment:
"XMLQuery" in  "pathToXMLQuery" could be mistaken for "XQuery", a technology for querying to contents of an XML document. Since it's a file, maybe rename to "pathToQueryFile"?